File: frontend/src/types/portal.ts
export interface PortalApp {
  id: string;
  name: string;
  description: string | null;
  status: string;
  icon?: string;
  preview_port: number | null;
  badges: PortalAppBadges;
}

export interface PortalAppBadges {
  pending_tasks: number;
  alerts: number;
  new_items: number;
}

export interface PortalTask {
  id: string;
  app_id: string;
  app_name: string;
  title: string;
  description: string | null;
  status: "pending" | "claimed" | "completed" | "expired" | "escalated";
  assignee_type: "person" | "role" | "group";
  due_at: string | null;
  created_at: string;
  workflow_name?: string;
}

export interface PortalActivity {
  id: string;
  app_id: string;
  app_name: string;
  type: "task_completed" | "task_created" | "alert" | "workflow_event" | "data_change";
  title: string;
  description: string | null;
  actor_name?: string;
  created_at: string;
}

export interface PortalDashboardData {
  apps: PortalApp[];
  tasks: PortalTask[];
  activity: PortalActivity[];
}

export interface PortalStats {
  total_apps: number;
  active_apps: number;
  pending_tasks: number;
  total_versions: number;
  recent_activity_count: number;
}

export interface SuggestedApp {
  department_id: string;
  department_name: string;
  suggestion: string;
}

export interface PortalStatsResponse {
  stats: PortalStats;
  suggestions: SuggestedApp[];
}

export interface PortalAppsResponse {
  apps: PortalApp[];
  total: number;
}

export interface PortalTasksResponse {
  tasks: PortalTask[];
  total: number;
}

export interface PortalActivityResponse {
  activity: PortalActivity[];
  total: number;
}

/** Export format options */
export type ExportFormat = "zip" | "dockerfile" | "git";

export interface ExportOptions {
  format: ExportFormat;
  includeReadme: boolean;
  includeDockerCompose: boolean;
  includeEnvExample: boolean;
  gitRemoteUrl?: string;
  gitBranch?: string;
}

export interface ExportResult {
  format: ExportFormat;
  success: boolean;
  message: string;
  downloadUrl?: string;
}

/** Command palette */
export interface CommandItem {
  id: string;
  label: string;
  description?: string;
  category: "navigation" | "action" | "model" | "component" | "page" | "workflow" | "rule";
  icon?: string;
  shortcut?: string;
  action: () => void;
}

/** Onboarding */
export type OnboardingStep =
  | "welcome"
  | "create_org"
  | "add_people"
  | "create_app"
  | "preview"
  | "complete";

export interface OnboardingState {
  currentStep: OnboardingStep;
  completedSteps: OnboardingStep[];
  dismissed: boolean;
}

/** Monitoring */
export interface CostEntry {
  date: string;
  input_tokens: number;
  output_tokens: number;
  cost_usd: number;
  agent_type: string;
}

export interface ProjectCostSummary {
  project_id: string;
  total_cost_usd: number;
  total_input_tokens: number;
  total_output_tokens: number;
  daily: CostEntry[];
  by_agent: Record<string, { cost_usd: number; calls: number }>;
}

export interface ErrorLogEntry {
  id: string;
  project_id: string;
  type: "agent_failure" | "build_failure" | "preview_crash" | "validation_error";
  message: string;
  details: string | null;
  agent_type: string | null;
  resolved: boolean;
  created_at: string;
}

File: backend/schemas/portal.py
"""Pydantic schemas for portal, export, and monitoring endpoints."""

from pydantic import BaseModel


# ---------------------------------------------------------------------------
# Portal
# ---------------------------------------------------------------------------

class PortalAppBadges(BaseModel):
    pending_tasks: int = 0
    alerts: int = 0
    new_items: int = 0


class PortalApp(BaseModel):
    id: str
    name: str
    description: str | None = None
    status: str
    icon: str | None = None
    preview_port: int | None = None
    badges: PortalAppBadges = PortalAppBadges()


class PortalTask(BaseModel):
    id: str
    app_id: str
    app_name: str
    title: str
    description: str | None = None
    status: str = "pending"
    assignee_type: str = "person"
    due_at: str | None = None
    created_at: str
    workflow_name: str | None = None


class PortalActivity(BaseModel):
    id: str
    app_id: str
    app_name: str
    type: str
    title: str
    description: str | None = None
    actor_name: str | None = None
    created_at: str


class PortalDashboard(BaseModel):
    apps: list[PortalApp] = []
    tasks: list[PortalTask] = []
    activity: list[PortalActivity] = []


class PortalStats(BaseModel):
    total_apps: int = 0
    active_apps: int = 0
    pending_tasks: int = 0
    total_versions: int = 0
    recent_activity_count: int = 0


class SuggestedApp(BaseModel):
    department_id: str
    department_name: str
    suggestion: str


# ---------------------------------------------------------------------------
# Export
# ---------------------------------------------------------------------------

class ExportRequest(BaseModel):
    format: str = "zip"  # zip | dockerfile | git
    include_readme: bool = True
    include_docker_compose: bool = True
    include_env_example: bool = True
    git_remote_url: str | None = None
    git_branch: str | None = None


# ---------------------------------------------------------------------------
# Monitoring
# ---------------------------------------------------------------------------

class CostEntry(BaseModel):
    date: str
    input_tokens: int = 0
    output_tokens: int = 0
    cost_usd: float = 0.0
    agent_type: str = ""


class ProjectCostSummary(BaseModel):
    project_id: str
    total_cost_usd: float = 0.0
    total_input_tokens: int = 0
    total_output_tokens: int = 0
    daily: list[CostEntry] = []
    by_agent: dict[str, dict] = {}


class ErrorLogEntry(BaseModel):
    id: str
    project_id: str
    type: str
    message: str
    details: str | None = None
    agent_type: str | None = None
    resolved: bool = False
    created_at: str
